Library/Homebrew/dev-cmd/tap-new.rb

```diff
@@ -82,12 +80,72 @@ def tap_new
 
             - run: brew test-bot --only-formulae
               if: github.event_name == 'pull_request'
+
+            - name: Upload bottles as artifact
+              if: always() && github.event_name == 'pull_request'
+              uses: actions/upload-artifact@main
+              with:
+                name: bottles
+                path: '*.bottle.*'
+    YAML
+
+    actions_publish = <<~YAML
+      name: brew pr-pull
+      on:
+        workflow_dispatch:
+          inputs:
+            pull_request:
+              description: Pull request number
+              required: true
+      jobs:
+        pr-pull:
+          runs-on: ubuntu-latest
+          steps:
+            - name: Set up Homebrew
+              uses: Homebrew/actions/setup-homebrew@master
+
+            - name: Set up git
+              uses: Homebrew/actions/git-user-config@master
+
+            - name: Pull bottles
+              env:
+                HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+                PULL_REQUEST: ${{github.event.inputs.pull_request}}
+              run: brew pr-pull --debug --tap=$GITHUB_REPOSITORY $PULL_REQUEST
+
+            - name: Push commits
+              uses: Homebrew/actions/git-try-push@master
+              with:
+                token: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+
+            - name: Delete branch
+              uses: peter-evans/close-pull@v1
+              with:
+                token: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+                pull-request-number: ${{github.event.inputs.pull_request}}
+                delete-branch: true
     YAML
 
     (tap.path/".github/workflows").mkpath
-    write_path(tap, ".github/workflows/tests.yml", actions)
+    write_path(tap, ".github/workflows/tests.yml", actions_main)
+    write_path(tap, ".github/workflows/publish.yml", actions_publish)
+
     ohai "Created #{tap}"
-    puts tap.path.to_s
+    puts <<~EOS
+      #{tap.path}
+
+      If you want to be able to publish bottles for your formulae in this tap,
+      then you should create a GitHub Personal Token with `public_repo` permission
+      and add it to repository secrets as `HOMEBREW_GITHUB_API_TOKEN`.
+      Link to token creation:
+        https://github.com/settings/tokens/new?scopes=public_repo
+
+      When a pull request making changes to a formula (or formulae) becomes green
+      (all checks passed), then you can publish the built bottles.
+      To do so, run:
+        brew pr-publish --tap=#{tap} <pull_request_number>
+      or trigger the workflow via GitHub UI in Actions tab.
```

Comment:
I wonder if there's anyway way of making this triggered by e.g. a label being added or similar?

Reply:
Hmm. We can use pull_request_target event trigger for that, but I'm not sure if we could squeeze it into this publishing workflow. Maybe use only this trigger instead of workflow_dispatch?

Reply:
Yeh, I think the workflow dispatch might be unnecessary for non-maintainers?